Sources/Utility/Diagnostics.swift

```diff
@@ -70,5 +70,21 @@ extension DiagnosticsEngine {
         location: DiagnosticLocation = UnknownLocation.location
      ) {
         emit(data: convertible.diagnosticData, location: location)
-    }
+	}
+
+	/// Wrap a throwing closure, returning an optional value and
+	/// emitting any thrown errors.
+	///
+	/// - Parameters:
+	///     - closure: Closure to wrap.
+	/// - Returns: Returns the return value of the closure wrapped
+	///   into an optional. If the closure throws, nil is returned.
+	public func wrap<T>(_ closure: () throws -> T) -> T? {
+		do {
+			return try closure()
+		} catch {
+			emit(error)
+			return nil
+		}
```

Comment:
Yep, sorry: I edited from work's Xcode which is configured with tabs - will fix!